File: problem1/m.clj
(reduce + (filter (fn [x] (or (== (mod x 3) 0) (== (mod x 5) 0))) (range 10)))

(->>
  (range 1000)
  (filter (fn [x] (or (== (mod x 3) 0) (== (mod x 5) 0))))
  (reduce +))

File: problem1/m.rs
use std::iter::AdditiveIterator;

fn is_divisible_to_3_or_5(num: i32) -> bool {
    num % 3 == 0 || num % 5 == 0
}

fn main() {
    let result = range(0i32, 1000).filter(|&x| is_divisible_to_3_or_5(x)).sum();
    println!("Sum is: {}", result);
}
